In [49]:
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
#Load the dataset

df = pd.read_csv('/Users/sakshideokar/Documents/OIM7502_SP24/data/Wage.csv')

In [4]:
#Display the first 10 rows

print(df.head(10))

   year  age            maritl      race        education              region  \
0  2006   18  1. Never Married  1. White     1. < HS Grad  2. Middle Atlantic   
1  2004   24  1. Never Married  1. White  4. College Grad  2. Middle Atlantic   
2  2003   45        2. Married  1. White  3. Some College  2. Middle Atlantic   
3  2003   43        2. Married  3. Asian  4. College Grad  2. Middle Atlantic   
4  2005   50       4. Divorced  1. White       2. HS Grad  2. Middle Atlantic   
5  2008   54        2. Married  1. White  4. College Grad  2. Middle Atlantic   
6  2009   44        2. Married  4. Other  3. Some College  2. Middle Atlantic   
7  2008   30  1. Never Married  3. Asian  3. Some College  2. Middle Atlantic   
8  2006   41  1. Never Married  2. Black  3. Some College  2. Middle Atlantic   
9  2004   52        2. Married  1. White       2. HS Grad  2. Middle Atlantic   

         jobclass          health health_ins   logwage        wage  
0   1. Industrial       1. <=Good      

In [5]:
#Display the last 10 rows

print(df.tail(10))

      year  age            maritl      race        education  \
2990  2009   50        2. Married  2. Black       2. HS Grad   
2991  2003   26  1. Never Married  2. Black  3. Some College   
2992  2007   35        2. Married  1. White     1. < HS Grad   
2993  2006   31        2. Married  1. White       2. HS Grad   
2994  2009   31        2. Married  1. White  4. College Grad   
2995  2008   44        2. Married  1. White  3. Some College   
2996  2007   30        2. Married  1. White       2. HS Grad   
2997  2005   27        2. Married  2. Black     1. < HS Grad   
2998  2005   27  1. Never Married  1. White  3. Some College   
2999  2009   55      5. Separated  1. White       2. HS Grad   

                  region        jobclass          health health_ins   logwage  \
2990  2. Middle Atlantic   1. Industrial  2. >=Very Good      2. No  4.886491   
2991  2. Middle Atlantic   1. Industrial  2. >=Very Good      2. No  4.778151   
2992  2. Middle Atlantic   1. Industrial       1. <=

In [6]:
#How many rows are in the DataFrame

number_Of_Rows = len(df)
print("The number of rows are", number_Of_Rows)

The number of rows are 3000


In [7]:
#Display metadata for the DataFrame

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   year        3000 non-null   int64  
 1   age         3000 non-null   int64  
 2   maritl      3000 non-null   object 
 3   race        3000 non-null   object 
 4   education   3000 non-null   object 
 5   region      3000 non-null   object 
 6   jobclass    3000 non-null   object 
 7   health      3000 non-null   object 
 8   health_ins  3000 non-null   object 
 9   logwage     3000 non-null   float64
 10  wage        3000 non-null   float64
dtypes: float64(2), int64(2), object(7)
memory usage: 257.9+ KB


In [8]:
Columns = df.columns
print('Column Names:', Columns)

Column Names: Index(['year', 'age', 'maritl', 'race', 'education', 'region', 'jobclass',
       'health', 'health_ins', 'logwage', 'wage'],
      dtype='object')


In [9]:
data_types = df.dtypes
print('Data types of each column:')
print(data_types)

Data types of each column:
year            int64
age             int64
maritl         object
race           object
education      object
region         object
jobclass       object
health         object
health_ins     object
logwage       float64
wage          float64
dtype: object


In [10]:
#Display the data types for two columns from the dataset

Column_Names = ['year', 'region']
for column_name in Column_Names:
    dtype = df[column_name].dtype
    print(f'Data type of {column_name}: {dtype}')

Data type of year: int64
Data type of region: object


In [11]:
#Display the column names from the dataset

Columns = df.columns
print('Column Names:', Columns)

Column Names: Index(['year', 'age', 'maritl', 'race', 'education', 'region', 'jobclass',
       'health', 'health_ins', 'logwage', 'wage'],
      dtype='object')


In [12]:
#Display the DataFrame’s index

df.index

RangeIndex(start=0, stop=3000, step=1)

In [13]:
#How many individual elements are in the DataFrame?

Elements = len(df.index) * df.shape[1]
print("Number of Elements", Elements)

Number of Elements 33000


In [14]:
#How many columns and rows are there?

Number_Of_Columns_And_Rows = df.shape
print("Number Of Columns & Rows are", Number_Of_Columns_And_Rows)

Number Of Columns & Rows are (3000, 11)


In [15]:
#Display descriptive statistics for the wage column, and the age column

Statistics = df[["wage", "age"]].describe()
print("Statistics:", Statistics)

Statistics:               wage          age
count  3000.000000  3000.000000
mean    111.703608    42.414667
std      41.728595    11.542406
min      20.085537    18.000000
25%      85.383940    33.750000
50%     104.921507    42.000000
75%     128.680488    51.000000
max     318.342430    80.000000


In [16]:
#Display descriptive statistics for the wage and age columns, by each race

Statistics = df.groupby('race')[["wage", "age"]].describe()
    
print("Statistics:", Statistics)

Statistics:             wage                                                           \
           count        mean        std        min        25%         50%   
race                                                                        
1. White  2480.0  112.563666  41.733830  20.085537  86.695155  106.348256   
2. Black   293.0  101.601177  37.162486  23.274704  81.283253   94.072715   
3. Asian   190.0  120.288288  46.422514  34.607418  87.981033  115.375039   
4. Other    37.0   89.973326  29.153530  38.223583  71.319639   81.283253   

                                     age                                     \
                 75%         max   count       mean        std   min    25%   
race                                                                          
1. White  130.042934  314.329336  2480.0  42.394758  11.365655  18.0  34.00   
2. Black  117.146817  318.342430   293.0  43.563140  13.012913  18.0  33.00   
3. Asian  144.828129  299.262977   190.0  41.821053  

In [18]:
#What do the mean and median values of wage tell us about the data?

df.wage.mean()

111.70360820174345

In [19]:
df.wage.median()

104.921506533664

The average wage for people in this data is 111.70 while the value of median tells us that out of 2999 observations, half of the people have wages less than 104.92. 

In [28]:
#Display the age and wage columns only

Column_names = df[['age', 'wage']]
Column_names

age        wage
0      18   75.043154
1      24   70.476020
2      45  130.982177
3      43  154.685293
4      50   75.043154
...   ...         ...
2995   44  154.685293
2996   30   99.689464
2997   27   66.229408
2998   27   87.981033
2999   55   90.481913

[3000 rows x 2 columns]

In [30]:
#Are these two columns correlated?

Correlation = df['age'].corr(df['wage'])
Correlation

0.1956372015635886

Positive value here shows that these 2 columns have a correlation that with increase in age, the wage also increases

In [40]:
#Display the top 10 salaries

Top_10_Salaries = df.sort_values(by = 'wage' , ascending = False)['wage']. head(10)
print("Top 10 salaries are:", Top_10_Salaries)

Top 10 salaries are: 206     318.342430
1229    318.342430
503     314.329336
1293    311.934569
2712    309.571771
1112    309.571771
2893    299.262977
2925    295.991250
489     284.524741
347     281.745971
Name: wage, dtype: float64


In [44]:
#Display the counts of each race in the dataset

race_count = {}
for race in df['race']:
    if race in race_count:
        race_count[race] += 1
    else:
        race_count[race] = 1

Each_Race_Count = pd.DataFrame(list(race_count.items()), columns=['race', 'count'])

print(race_counts_df)

       race  count
0  1. White   2480
1  3. Asian    190
2  4. Other     37
3  2. Black    293


In [48]:
#Display the proportion of each race in the dataset

Total_Records = Each_Race_Count['count'].sum()
Each_Race_Count['proportion'] = Each_Race_Count['count'] / Total_Records
print("Proportion of each race in the dataset:", Each_Race_Count)



Proportion of each race in the dataset:        race  count  proportion
0  1. White   2480    0.826667
1  3. Asian    190    0.063333
2  4. Other     37    0.012333
3  2. Black    293    0.097667


In [ ]:
#Create a bar plot of regional proportionality

#I couldn't get this to work